src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -0,0 +1,615 @@
+ * will result in a split outcome encoded in the returned mutation_result_t
+ */
+OMapInnerNode::make_split_insert_ret
+OMapInnerNode::make_split_insert(omap_context_t oc, internal_iterator_t iter,
+                                 std::string key, laddr_t laddr)
+{
+  return make_split_children(oc).safe_then([=] (auto tuple) {
+    auto [left, right, pivot] = tuple;
+    if (pivot > key) {
+      auto liter = left->iter_idx(iter.get_index());
+      left->journal_inner_insert(liter, laddr, key,
+                                 left->maybe_get_delta_buffer());
+    } else {  //right
+      auto riter = right->iter_idx(iter.get_index() - left->get_node_size());
+      right->journal_inner_insert(riter, laddr, key,
+                                  right->maybe_get_delta_buffer());
+    }
+    return make_split_insert_ret(
+           make_split_insert_ertr::ready_future_marker{},
+           mutation_result_t(mutation_status_t::SPLITTED, tuple, std::nullopt));
+  });
+
+}
+
+
+OMapInnerNode::handle_split_ret
+OMapInnerNode::handle_split(omap_context_t oc, internal_iterator_t iter,
+                               mutation_result_t mresult)
+{
+  logger().debug("{}: {}","OMapInnerNode",  __func__);
+  if (!is_pending()) {
+    auto mut = oc.tm.get_mutable_extent(oc.t, this)->cast<OMapInnerNode>();
+    auto mut_iter = mut->iter_idx(iter.get_index());
+    return mut->handle_split(oc, mut_iter, mresult);
+  }
+  auto [left, right, pivot] = *(mresult.split_tuple);
+  //update will not cause overflow do it first.
+  journal_inner_update(iter, left->get_laddr(), maybe_get_delta_buffer());
+  if (!extent_will_overflow(pivot.size() + 1, std::nullopt)) {
+    journal_inner_insert(iter + 1, right->get_laddr(), pivot,
+                         maybe_get_delta_buffer());
+    return insert_ret(
+           insert_ertr::ready_future_marker{},
+           mutation_result_t(mutation_status_t::SUCCESS, std::nullopt, std::nullopt));
+  } else {
+    return make_split_insert(oc, iter + 1, pivot, right->get_laddr())
+      .safe_then([this, oc] (auto m_result) {
+       return oc.tm.dec_ref(oc.t, get_laddr())
+         .safe_then([m_result = std::move(m_result)] (auto ret) {
+          return insert_ret(
+                 insert_ertr::ready_future_marker{},
+                 m_result);
+       });
+   });
+  }
+}
+
+OMapInnerNode::get_value_ret
+OMapInnerNode::get_value(omap_context_t oc, const std::string &key)
+{
+  logger().debug("{}: {} key = {}", "OMapInnerNode", __func__, key);
+  auto child_pt = get_containing_child(key);
+  auto laddr = child_pt->get_node_key().laddr;
+  return omap_load_extent(oc, laddr, get_meta().depth - 1).safe_then(
+    [oc, &key] (auto extent) {
+    return extent->get_value(oc, key);
+  }).finally([ref = OMapNodeRef(this)] {});
+}
+
+OMapInnerNode::insert_ret
+OMapInnerNode::insert(omap_context_t oc, const std::string &key, const std::string &value)
+{
+  logger().debug("{}: {}  {}->{}", "OMapInnerNode",  __func__, key, value);
```

Comment:
move the "OmapInner..." to the format string?

Reply:
is this using fmt?  that's cool